manager/handlers/application_test.go

```diff
@@ -0,0 +1,348 @@
+	mockPriorityValue            = 20
+	mockCreateApplicationRequest = types.CreateApplicationRequest{
+		Name:   "foo",
+		URL:    "http://example.com/foo",
+		BIO:    "bio",
+		UserID: 4,
+		Priority: &types.PriorityConfig{
+			Value: &mockPriorityValue,
+			URLs: []types.URLPriorityConfig{
+				{
+					Regex: "regex",
+					Value: 20,
+				},
+			},
+		},
+	}
+	mockUpdateApplicationRequest = types.UpdateApplicationRequest{
+		Name:   "foo",
+		URL:    "http://example.com/foo",
+		BIO:    "bio",
+		UserID: 4,
+		Priority: &types.PriorityConfig{
+			Value: &mockPriorityValue,
+			URLs: []types.URLPriorityConfig{
+				{
+					Regex: "regex",
+					Value: 20,
+				},
+			},
+		},
+	}
+	mockBaseModel = models.BaseModel{
+		ID:        2,
+		CreatedAt: time.Date(2024, 4, 17, 18, 0, 21, 580470900, time.UTC),
+		UpdatedAt: time.Date(2024, 4, 17, 18, 0, 21, 580470900, time.UTC),
+		IsDel:     soft_delete.DeletedAt(soft_delete.FlagActived),
+	}
+	mockApplicationModel = &models.Application{
+		BaseModel: mockBaseModel,
+		Name:      "foo",
+		URL:       "http://example.com/foo",
+		BIO:       "bio",
+		UserID:    4,
+		Priority:  models.JSONMap{"value": 20, "urls": []interface{}{map[string]interface{}{"regex": "regex", "value": 20}}},
+	}
+)
+
+func mockApplicationRouter(h *Handlers) *gin.Engine {
+	r := gin.Default()
+	apiv1 := r.Group("/api/v1")
+	cs := apiv1.Group("/applications")
+	cs.POST("", h.CreateApplication)
+	cs.DELETE(":id", h.DestroyApplication)
+	cs.PATCH(":id", h.UpdateApplication)
+	cs.GET(":id", h.GetApplication)
+	cs.GET("", h.GetApplications)
+	return r
+}
+
+func TestHandlers_CreateApplication(t *testing.T) {
+	tests := []struct {
+		name   string
+		req    *http.Request
+		mock   func(ms *mocks.MockServiceMockRecorder)
+		expect func(t *testing.T, w *httptest.ResponseRecorder)
+	}{
+		{
+			name: "unprocessable entity",
+			req:  httptest.NewRequest("POST", "/api/v1/applications", nil),
+			mock: func(ms *mocks.MockServiceMockRecorder) {},
+			expect: func(t *testing.T, w *httptest.ResponseRecorder) {
+				assert := assert.New(t)
+				assert.Equal(http.StatusUnprocessableEntity, w.Code)
+			},
+		},
+		{
+			name: "success",
+			req:  httptest.NewRequest("POST", "/api/v1/applications", strings.NewReader(mockApplicationReqBody)),
+			mock: func(ms *mocks.MockServiceMockRecorder) {
+				ms.CreateApplication(gomock.Any(), gomock.Eq(mockCreateApplicationRequest)).Return(mockApplicationModel, nil).Times(1)
+			},
+			expect: func(t *testing.T, w *httptest.ResponseRecorder) {
+				assert := assert.New(t)
+				assert.Equal(http.StatusOK, w.Code)
+				assert.Equal(w.Body.String(), mockApplicationResponseBody)
```

Comment:
Unmarshal `w.Body.String()` to struct, remove `mockApplicationReqBody`.

```golang
application := models.Application{}
if err := json.Unmarshal(...) {
...
}

assert.Equal(resp, models.Application{...})
```

Please modify the same type of cases.